common/configuration/db_config_provider.go

```diff
@@ -0,0 +1,50 @@
+package configuration
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/TBD54566975/ftl/backend/controller/dalerrors"
+	"github.com/TBD54566975/ftl/backend/controller/sql"
+)
+
+// DBConfigProvider is a configuration provider that stores configuration in its key.
+type DBConfigProvider struct {
+	DB    bool `help:"Write configuration values to the database." group:"Provider:" xor:"configwriter"`
+	dalDB sql.DBI
+}
+
+var _ MutableProvider[Configuration] = DBConfigProvider{}
+
+func NewDBConfigProvider(db sql.DBI) DBConfigProvider {
```

Comment:
instead of using `db sql.DBI` here and exposing direct access to the `db` we should probably just use the `interface` as @alecthomas suggested. If I'm reading his comment correctly, that might end up looking similar to the approach here in [cronjobs.go](https://github.com/TBD54566975/ftl/blob/main/backend/controller/cronjobs/cronjobs.go#L68-L73).

Then, your new stuff can use your new `DAL` which would only expose the methods needed by your new provider. It will likely make things a bit easier to test as well. 👍

Reply:
done :)